packages/web3/src/contract/ralph.ts

```diff
@@ -385,6 +385,28 @@ export function primitiveToByteVec(value: Val, type: string): Uint8Array {
   }
 }
 
+export function typeLength(typ: string, structs: Struct[]): number {
+  if (isPrimitiveType(typ)) {
+    return 1
+  }
+
+  if (typ.startsWith('[')) {
+    const [baseType, size] = decodeArrayType(typ)
+    return size * typeLength(baseType, structs)
+  }
+
+  const struct = structs.find((s) => s.name === typ)
+  if (struct !== undefined) {
+    return struct.fieldTypes.reduce((acc, fieldType) => acc + typeLength(fieldType, structs), 0)
+  }
+
+  throw Error(`Invalid type ${typ}`)
+}
+
+export function isPrimitiveType(typ: string): boolean {
+  return ['Bool', 'I256', 'U256', 'ByteVec', 'Address'].includes(typ)
```

Comment:
We have already defined the primitive types here: https://github.com/alephium/alephium-web3/blob/1325c8cb7a071820bcb2ea3d92539d2d2bebaa32/packages/web3/src/api/types.ts#L30
maybe we can reuse this one.